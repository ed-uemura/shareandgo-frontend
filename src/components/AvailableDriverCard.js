import React from 'react';
import { Link } from 'react-router-dom';
import RatingStars from './RatingStars';

const AvailableDriverCard = ({
	driverPhoto,
	driverName,
	driverRating,
	destination,
	availableSeats,
	date,
	time,
	price,
	link,
}) => {
	return (
		<Link to={link}>
			<div className='ride-card'>
				<div className='photo'>
					<figure className='driver-picture'>
						<img src={driverPhoto} alt={driverName} />
					</figure>
				</div>
				<div className='driver-info'>
					<p className='text-important'>{driverName}</p>

					<RatingStars rating={driverRating} />

					<div className='destination d-flex align-items-center'>
						<span className='material-symbols-outlined card-icon me-2'>
							directions_car
						</span>
						<p className='details'>{destination}</p>
					</div>
					<div className='seats-available d-flex align-items-center'>
						<span className='material-symbols-outlined card-icon me-2'>
							person
						</span>
						<p className='details'>{availableSeats} seats available</p>
					</div>
				</div>
				<div className='ride-info'>
					<p className='ride-details'>{date}</p>
					<p className='ride-details'>{time}</p>
					<br></br>
					<p className='text-important'>${price}</p>
				</div>
			</div>
		</Link>
	);
};

export default AvailableDriverCard;
