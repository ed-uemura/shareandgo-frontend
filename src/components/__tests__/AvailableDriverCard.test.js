import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import AvailableDriverCard from '../AvailableDriverCard';
import DriverImage from '../../images/profile-picture-mockup.jpg';
import { MemoryRouter } from 'react-router-dom';


afterEach(() =>{
    cleanup();
});

test('matches snapshot', () =>{ 
    const testInfo ={
    driverPhoto : DriverImage,
	driverName : 'Mathew Pitts',
	destination : 'Abbotsford, BC',
	date : '10/15/2022',
	time : '9:45am',
	link : '/message-content',
	driverRating : '4',
	availableSeats : '3',
	price : '128.95',
	link : '/ride-info',
    }

    const tree = renderer.create(
        <MemoryRouter>
            <AvailableDriverCard {...testInfo}/>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});