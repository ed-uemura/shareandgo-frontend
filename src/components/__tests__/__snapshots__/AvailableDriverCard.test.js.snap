// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  href="/ride-info"
  onClick={[Function]}
>
  <div
    className="ride-card"
  >
    <div
      className="photo"
    >
      <figure
        className="driver-picture"
      >
        <img
          alt="Mathew Pitts"
          src="profile-picture-mockup.jpg"
        />
      </figure>
    </div>
    <div
      className="driver-info"
    >
      <p
        className="text-important"
      >
        Mathew Pitts
      </p>
      <div>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={0}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={1}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={2}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={3}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined"
            id={4}
          >
            star
          </span>
        </i>
      </div>
      <div
        className="destination d-flex align-items-center"
      >
        <span
          className="material-symbols-outlined card-icon me-2"
        >
          directions_car
        </span>
        <p
          className="details"
        >
          Abbotsford, BC
        </p>
      </div>
      <div
        className="seats-available d-flex align-items-center"
      >
        <span
          className="material-symbols-outlined card-icon me-2"
        >
          person
        </span>
        <p
          className="details"
        >
          3
           seats available
        </p>
      </div>
    </div>
    <div
      className="ride-info"
    >
      <p
        className="ride-details"
      >
        10/15/2022
      </p>
      <p
        className="ride-details"
      >
        9:45am
      </p>
      <br />
      <p
        className="text-important"
      >
        $
        128.95
      </p>
    </div>
  </div>
</a>
`;
